lib/galaxy/jobs/runners/util/cli/shell/rsh.py

```diff
@@ -37,15 +40,13 @@ class SecureShell(RemoteShell):
 
     def __init__(self, rsh='ssh', rcp='scp', private_key=None, port=None, strict_host_key_checking=True, **kwargs):
         strict_host_key_checking = "yes" if strict_host_key_checking else "no"
-        rsh += " -oStrictHostKeyChecking=%s -oConnectTimeout=60" % strict_host_key_checking
-        rcp += " -oStrictHostKeyChecking=%s -oConnectTimeout=60" % strict_host_key_checking
+        options = ["-oStrictHostKeyChecking=%s" % strict_host_key_checking]
+        options.append("-oConnectTimeout=60")
         if private_key:
-            rsh += " -i %s" % private_key
-            rcp += " -i %s" % private_key
+            options.extend(['-i', private_key])
         if port:
-            rsh += " -p %s" % port
-            rcp += " -p %s" % port
-        super(SecureShell, self).__init__(rsh=rsh, rcp=rcp, **kwargs)
+            options.extend(['-p', str(port)])
```

Comment:
It is for the unit test, overall I think it's sensible for a `port` variable to accept integer (as paramiko for instance does).

Reply:
I mean that `Popen` does the casting already.

Reply:
```
In [3]: subprocess.Popen(['echo', 5])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-6bc2fd563ae2> in <module>()
----> 1 subprocess.Popen(['echo', 5])

/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.pyc in __init__(self, args, bufsize, executable, stdin, stdout, stderr, preexec_fn, close_fds, shell, cwd, env, universal_newlines, startupinfo, creationflags)
    392                                 p2cread, p2cwrite,
    393                                 c2pread, c2pwrite,
--> 394                                 errread, errwrite)
    395         except Exception:
    396             # Preserve original exception in case os.close raises.

/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.pyc in _execute_child(self, args, executable, preexec_fn, close_fds, cwd, env, universal_newlines, startupinfo, creationflags, shell, to_close, p2cread, p2cwrite, c2pread, c2pwrite, errread, errwrite)
   1045                         raise
   1046                 child_exception = pickle.loads(data)
-> 1047                 raise child_exception
   1048
   1049

TypeError: execv() arg 2 must contain only strings
```

Reply:
Ops, I assumed that was the case for what I remembered from lib/galaxy/jobs/runners/slurm.py , but evidently `job_id` must always be a string there. Sorry for the noise!

Reply:
No worries, definitely appreciate the review.